org.adempiere.base/src/org/compiere/model/MTable.java

```diff
@@ -667,7 +667,7 @@ public PO getUUIDPOInstance (String uuid, String trxName)
 		if (cache != null)
 		{
 			IModelFactory service = cache.getService();
-			if (service != null)
+			if (service != null && service.getClass(tableName) != null)

```

Comment:
this is redundant, service.getPO below should return null, otherwise it is a violation of the API contract.